### PR TITLE
Add `Style/QueryBoolMethods` rule

### PR DESCRIPTION
**Pull request by Sija**
Reports boolean properties defined with `?`-less variants of `property` / `getter` macros.

```cr
class Person
  property deceased = false
  getter witty = true
end
```

### CODE REVIEW

File: src/ameba/ast/util.cr [hunk 1]
```diff
@@ -58,6 +58,20 @@ module Ameba::AST::Util
...
+  def path_named?(node, name) : Bool
+    case node
+    when Crystal::Path
+      node_name = node.names.join("::")
+      node_name = "::#{node_name}" if node.global?
+
+      name == node_name
+    else
+      false
+    end
```
Comment (veelenga) [inline src/ameba/ast/util.cr:72]: case statement seems to be redundant here:

```suggestion
    return false unless node.is_a?(Crystal::Path)

    node_name = node.names.join("::")
    node_name = "::#{node_name}" if node.global?

    name == node_name
```

File: spec/ameba/rule/style/query_bool_methods_spec.cr [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          class Foo
+            %{call}   foo : Bool
+            _{call} # ^ error: Consider using '%{call}?' for 'foo'
+
+            def initialize(@foo = true)
+            end
+          end
+          CRYSTAL
+      end
+    {% end %}
```
Comment (veelenga) [inline spec/ameba/rule/style/query_bool_methods_spec.cr:58]: please add a test that checks if there are two boolean properties and only one is invalid

Comment (Sija) [reply spec/ameba/rule/style/query_bool_methods_spec.cr:58]: I don't think I understand what do you mean.

Comment (Sija) [reply spec/ameba/rule/style/query_bool_methods_spec.cr:58]: I still don't get it... Adjacent properties do not affect each other...

Comment (veelenga) [reply spec/ameba/rule/style/query_bool_methods_spec.cr:58]: > Adjacent properties do not affect each other

That actually highly depends on the implementation. Here is an example:

```diff
diff --git a/src/ameba/rule/style/query_bool_methods.cr b/src/ameba/rule/style/query_bool_methods.cr
index 4bd8c370..0750b143 100644
--- a/src/ameba/rule/style/query_bool_methods.cr
+++ b/src/ameba/rule/style/query_bool_methods.cr
@@ -66,6 +66,7 @@ module Ameba::Rule::Style
             issue_for name_node, MSG % {exp.name, name_node}
           end
         end
+        return
       end
     end
   end
```

if i slightly change the code, the test above fails, but all other pass

**Review by veelenga — APPROVED** [review]
LGTM

**Pull request merged**